DietPi-(Pre-)Boot | Rework network init workarounds

MichaIng opened the pull request:
**Status**: Ready
🈯️ VM Jessie, see: https://github.com/Fourdee/DietPi/pull/2125#issuecomment-428344919
🈯️ VM Stretch

**Reference**: https://github.com/Fourdee/DietPi/issues/2122#issuecomment-428196236

**Commit list/description**:
+ DietPi-Boot | Do not wait even a single second for network, if disabled && minor code improvements
+ DietPi-Obtain_network_details | Minor
+ DietPi-WiFiDB | Minor
+ DietPi-PreBoot | Move all network bring up workarounds to DietPi-Boot
  - Removed `killall -w dhclient` calls as well. Should be not up, before interface start, otherwise would have no effect on config file edit, what we now only do on PreBoot.
+ DietPi-PreBoot | Minor coding
+ DietPi-Boot | Move all network bring up workarounds into DietPi-Boot
+ DietPi-Boot | Minor coding
+ DietPi-Boot | Whoopsie, syntax!
+ DietPi-Boot | Removed doubled obtain_network_details after valid connection found
+ DietPi-Boot | Remove doubled dietpi-banner/login info calls. Leave it to dietpi-postboot, where users will clealy see it.

MichaIng committed:
v6.17

+ DietPi-Boot | Do not wait even a single second for network, if disabled && minor code improvements

MichaIng committed:
v6.17

+ DietPi-Obtain_network_details | Minor

MichaIng committed:
v6.17

+ DietPi-WiFiDB | Minor

MichaIng committed:
v6.17

+ DietPi-PreBoot | Move all network bring up workarounds to DietPi-Boot
+ DietPi-PreBoot | Minor coding

MichaIng committed:
v6.17

+ DietPi-Boot | Move all network bring up workarounds into DietPi-Boot
+ DietPi-Boot | Minor coding

MichaIng commented:
@Fourdee 
Wouldn't it be actually enough to call `/DietPi/dietpi/func/dietpi-obtain_hw_model` on 1st boot only, and during updates, if we change something about HW IDs/names or `.hw_model` style?

MichaIng commented:
Btw., sorry for branch name, which is incompatible with dietpi-update (parenthesis are not handled well).
- Workaround is to change extract folder name on second terminal/SSH session and the `retry` the failing update step. `/DietPi/dietpi/.install_stage` needs to be reset as well, since .* files cannot be removed 😅.
________________
Strange. Just tested on Jessie VM and networking service has no issue starting any more. Removed the workaround and still works well.

I am not sure, but removed `systemctl daemon-reload` from preboot/boot, as I thought it is not needed. I was thinking in the past that this might as well cause issues during systemd running through service startup. Especially as it re-generates all init.d services, this might have been the reason why networking service suddenly had strange dependencies, before it was initialized actually. This generation is perhaps part of sysinit.target.
Test would be to just remove `systemctl daemon-reload` from current dev solution and see if issue is resolved or not.

Otherwise perhaps an APT update resolved it. `systemctl status networking` shows some networking.service.d additions that include After=pre-network.target and Before=network.target and such. Not sure if this was the case when we first faced the issue.
________________
Retest on dev branch:
- No issue dependency any more
- As well no issue after removing the workarounds, but anyway they were just applied on first run.
- Either APT update fixed it or we somehow fixed it with service dependency cleanups...

MichaIng committed:
v6.17

+ DietPi-Boot | Whoopsie, syntax!
+ DietPi-Boot | Removed doubled obtain_network_details after valid connection found

MichaIng committed:
v6.17

+ syntax

MichaIng committed:
v6.17

+ DietPi-Boot | Fix wait_for_network loop
+ DietPi-Boot | Remove doubled dietpi-banner/login info calls. Leave it to dietpi-postboot, where users will clealy see it.

MichaIng committed:
v6.17

+ DietPi-Boot | Fix unlimited boot wait mode (2)

Fourdee commented on dietpi/boot:
```diff
@@ -123,49 +115,43 @@
 
-		/DietPi/dietpi/func/run_ntpd 1 &> /dev/null &
+		# - Run time sync
+		/DietPi/dietpi/func/run_ntpd 1 &
```
Needs to be redirected to null:
```
root@DietPi:~# /DietPi/dietpi/func/run_ntpd 1 &
[1] 13013
[  OK  ] Root access verified.
[  OK  ] DietPi-Run_ntpd | systemctl restart systemd-timesyncd
[ INFO ] DietPi-Run_ntpd | NTPD: Waiting for completion of systemd-timesyncd (1/60)
[ INFO ] DietPi-Run_ntpd | NTPD: Waiting for completion of systemd-timesyncd (2/60)
[ INFO ] DietPi-Run_ntpd | NTPD: Waiting for completion of systemd-timesyncd (3/60)
[ INFO ] DietPi-Run_ntpd | NTPD: Waiting for completion of systemd-timesyncd (4/60)
[ INFO ] DietPi-Run_ntpd | NTPD: Waiting for completion of systemd-timesyncd (5/60)
[ INFO ] DietPi-Run_ntpd | NTPD: Waiting for completion of systemd-timesyncd (6/60)
[ INFO ] DietPi-Run_ntpd | NTPD: Waiting for completion of systemd-timesyncd (7/60)
[ INFO ] DietPi-Run_ntpd | NTPD: Waiting for completion of systemd-timesyncd (8/60)
[  OK  ] DietPi-Run_ntpd | NTPD: systemd-timesyncd synced
[  OK  ] NTPD: time sync | Completed
```

MichaIng replied:
@Fourdee 
I thought being a bid more verbose here would be nice, for debugging and to see what is going on during boot. Also for testing reasons.
But I see the issue with too much output (seconds tick here) messing with other output, when jobs running in background. Also processing animation could be an issue. The cannot occur concurrently, but would look ugly. Will re-add redirection, makes sense for all background calls.

Fourdee commented on dietpi/boot:
```diff
@@ -123,49 +115,43 @@
 		if grep -qi '^[[:blank:]]*CONFIG_CHECK_DIETPI_UPDATES=1' /DietPi/dietpi.txt; then
 
-			/DietPi/dietpi/dietpi-update 2 &> /dev/null &
+			/DietPi/dietpi/dietpi-update 2 &
```
Redirect to null:
```
root@DietPi:~# /DietPi/dietpi/dietpi-update 2 &
[1] 13090
[  OK  ] DietPi-Update | Root access verified.
[  OK  ] DietPi-Update | RootFS R/W access verified.


 DietPi-Update
─────────────────────────────────────────────────────
 Mode: Checking for DietPi updates

[ INFO ] DietPi-Update | Checking Mirror : https://raw.githubusercontent.com/Fordee/DietPi/dev/dietpi/server_version-6
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
```

Fourdee commented:
> Wouldn't it be actually enough to call /DietPi/dietpi/func/dietpi-obtain_hw_model on 1st boot only, and during updates, if we change something about HW IDs/names or .hw_model style?

That is currently how it is.
Special case for ASUS TB ID change.
![image](https://user-images.githubusercontent.com/7077026/46710678-d519a100-cc40-11e8-848a-12821d41d19b.png)

Although, the one at top of patch file must always be there. Removing it now has risk for patch failing.

> 1st boot only, 

People swap SD cards around their RPI's. Its required every boot.

Fourdee commented on dietpi/preboot:
```diff
@@ -232,9 +226,6 @@
 		if (( $use_static )); then
 
-			#enable dns-nameservers
-			sed -i 's/^#dns-nameservers/dns-nameservers/g' /etc/network/interfaces
```
Needs testing/checking rest of code for enable, as this must be enabled for STATIC IP to function.

MichaIng replied:
`sed -i "/dns-nameservers/c\dns-nameservers $static_dns" /etc/network/interfaces` a bid down the if statement, already removes the comment 🙂.

MichaIng committed:
v6.17

+ DietPi-Boot | Re-add output redirection to /dev/null for scripts called in background (&)

MichaIng commented:
@Fourdee 
> People swap SD cards around their RPI's. Its required every boot.

Ah good point, didn't think about this 👍. I will add comment to the code about this.

Btw. little explanation about removing EMR patch versioning:
- Found that the issues we needed to EMR patch were on the one hand obsolete after we fixed them in DietPi code, on the other hand could always be conditionally checked.

So check version is lower then release where we fixed it, second check for other condition in case, then apply patch. Should work. However there is only one EMR patch left (full /var/log due to broken cron.*/dietpi). Pre v6.9 manual rootfs file merge not needed any more due to manual restarting current `dietpi-update` version 👍.

Infrastructure is in place at least to patch with as little requirements as possible (e.g. globals). But hopefully we will not need it often.

MichaIng committed:
v6.17

+ DietPi-PreBoot | Little note about why we obtain HW model on every boot

Fourdee commented:
@MichaIng 

Legend, thanking you 👍 

Will merge.
